packages/react-core/src/Authenticator/hooks/utils.ts

```diff
@@ -0,0 +1,43 @@
+import { AuthenticatorRoute } from '@aws-amplify/ui';
+
+import { COMPONENT_ROUTE_KEYS, COMPONENT_ROUTE_NAMES } from './constants';
+import {
+  AuthenticatorRouteComponentKey,
+  Defaults,
+  DefaultComponent,
+  Overrides,
+} from './types';
+
+export const isComponentRouteKey = (
+  route: AuthenticatorRoute
+): route is AuthenticatorRouteComponentKey =>
+  COMPONENT_ROUTE_KEYS.some((componentRoute) => componentRoute === route);
+
+export function resolveAuthenticatorComponents<PlatformProps = {}>(
```

Comment:
This function may seem tangential to `useAuthenticatorRoute`, but it is included as it is the intermediary function between user defined `Authenticator` sub-components and `useAuthenticatorRoute`. Any updates to `useAuthenticatorRoute` may have potential impact to `resolveAuthenticatorComponents` (and vice versa) so included it in this PR